support/ci/promote_rc.sh

```diff
@@ -2,33 +2,33 @@
 
 set -e
 
-if [ -z ${HAB_AUTH_TOKEN+x} ]; then
+if [ -z "${HAB_AUTH_TOKEN+x}" ]; then
   echo "HAB_AUTH_TOKEN var is unset"
   exit 1
 fi
-if [ -z ${BINTRAY_USER+x} ]; then
+if [ -z "${BINTRAY_USER+x}" ]; then
   echo "BINTRAY_USER var is unset"
   exit 1
 fi
-if [ -z ${BINTRAY_KEY+x} ]; then
+if [ -z "${BINTRAY_KEY+x}" ]; then
   echo "BINTRAY_KEY var is unset"
   exit 1
 fi
 
 HAB_VERSION=$(cat VERSION)
 PROMOTE_CHANNEL=stable
 RC_CHANNEL=rc-$HAB_VERSION
-RC_PKGS_JSON=$(curl https://bldr.habitat.sh/v1/depot/channels/core/$RC_CHANNEL/pkgs)
-RC_PKGS_JSON_DATA=$(echo $RC_PKGS_JSON | jq -c -r '.data | .[]')
-TOTAL_COUNT=$(echo $RC_PKGS_JSON | jq -r '.total_count')
-END=$(echo $RC_PKGS_JSON | jq -r '.range_end')
+RC_PKGS_JSON=$(curl https://bldr.habitat.sh/v1/depot/channels/core/"$RC_CHANNEL"/pkgs)
+RC_PKGS_JSON_DATA=$(echo "$RC_PKGS_JSON" | jq -c -r '.data | .[]')
+TOTAL_COUNT=$(echo "$RC_PKGS_JSON" | jq -r '.total_count')
+END=$(echo "$RC_PKGS_JSON" | jq -r '.range_end')
 
 # We limit results to 50 items. While it is unlikely, it is possible
 # after building several releases that we may exceed 50 packages in
 # the rc channel. We could get all fancy and complicated here to
 # aggregate multiple pages but I'd rather tear my eyes out of their
 # sockets. Instead, we suggest a workaround in the failure message.
-if [ $(($END+1)) !=  $TOTAL_COUNT ]; then
+if [ $((END+1)) !=  TOTAL_COUNT ]; then
```

Comment:
See https://github.com/koalaman/shellcheck/wiki/SC2004#-is-unnecessary-on-arithmetic-variables